libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Bot.Builder.Integration.AspNet.Core
+{
+    /// <summary>
+    /// Defines extension methods for to add common services to the application's service collection<see cref="IServiceCollection"/>.
+    /// </summary>
+    public static class ServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds bot related services to the application's service collection.
+        /// </summary>
+        /// <remark>
+        /// The following dependencies are added with TrySingleton so advanced scenarios can override them to customize the runtime behavior:
+        /// <see cref="BotFrameworkAuthentication"/>,
+        /// <see cref="IBotFrameworkHttpAdapter"/>,
+        /// <see cref="IStorage"/>,
+        /// <see cref="UserState"/>,
+        /// <see cref="ConversationState"/>
+        /// 
+        /// This set of dependencies is designed to be sufficient to run a typical bot. Since each of these
+        /// are registered using TrySingleton to provide a different implementation of any of the dependencies
+        /// just register them with the service collection before calling AddBotRuntime.
+        /// 
+        /// </remark>
+        /// <param name="services">The application's collection of registered services.</param>
+        /// <param name="configuration">The application configuration.</param>
+        public static void AddBotRuntime(this IServiceCollection services, IConfiguration configuration)
+        {
+            RegisterCommonServices(services, configuration);
+        }
+
+        /// <summary>
+        /// Overload that allows for creating the Bot (IBot) type as well as other startup types.
+        /// </summary>
+        /// <typeparam name="TBot">Type of Bot.</typeparam>
+        /// <param name="services">The application's collection of registered services.</param>
+        /// <param name="configuration">The application configuration.</param>
+        public static void AddBotRuntime<TBot>(this IServiceCollection services, IConfiguration configuration) 
+            where TBot : class, IBot
+        {
+            RegisterCommonServices(services, configuration);
+
+            services.AddSingleton<IBot, TBot>();
+        }
+
+        /// <summary>
+        /// Overload that allows for providing the Bot (IBot) type, and the main Dialog type as well as other startup types.
```

Comment:
Nits around the docs, since these are the main entry points for folks, it's worth spending a bit more time here. I like the phrasing of the main overload

`Adds bot related services to the application's service collection.`

plus all the remarks they have. Can we make the overloads have the same remark structure and summary (adapted to the overload specific fields / arguments)?

I think it's better to have all overloads have the same structure rather than having a highly documented one and then others referred to as overload. Otherwise people will see the remarks only on the top overload. If you check dotnetfx methods they don't speak of overloads, but they just document each method equally.

Reply:
Thanks, I've updated each of the constructor to have full comments.